File: countsetbit .java
/**
 * countsetbit 
 */
class countsetbit  {

    public static void main(String arg[])
    {
        int n=42;
        System.out.println("The number of set bits are: "+countsetbit.countSetBits(n));
    }
    public static int countSetBits(int n){
    
        // Your code here
        int res;
        for(res=0;n>0;n/=2)
            if(n%2==1)
                res++;
        return res;
    }
}

File: countsetbit.cpp
#include<iostream>
using namespace std;
int main()
{
    int n;
    cout<<"Enter the number: ";
    cin>>n;
    int res=0;
    while(n>0)
    {
        if(n%2==1)
            res++;
        n/=2;
    }
    cout<<"the number of set bits are: "<<res;
    return 0;
}
